container/container.go

```diff
@@ -914,6 +916,19 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 		}
 	}
 
+	var dns []string
+
+	if len(container.HostConfig.DNS) > 0 {
+		dns = container.HostConfig.DNS
+	} else if len(epConfig.DNS) > 0 {
+		dns = epConfig.DNS
```

Comment:
daemon is not available in container.go. I would have to pass it if I have to do that which will require changing the public function BuildCreateEndpointOptions. If that is alright I would do that